staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -546,24 +546,24 @@ func RunOptionalTestProgressNotify(ctx context.Context, t *testing.T, store stor
 
 	// when we send a bookmark event, the client expects the event to contain an
 	// object of the correct type, but with no fields set other than the resourceVersion
-	testCheckResultFunc(t, watch.Bookmark, w, func(object runtime.Object) error {
+	testCheckResultFunc(t, w, func(actualEvent watch.Event) {
+		expectObjectsDeepEqual(t, "incorrect event type", watch.Bookmark, actualEvent.Type)
 		// first, check that we have the correct resource version
-		obj, ok := object.(metav1.Object)
+		obj, ok := actualEvent.Object.(metav1.Object)
 		if !ok {
-			return fmt.Errorf("got %T, not metav1.Object", object)
+			t.Fatalf("got %T, not metav1.Object", actualEvent.Object)
```

Comment:
t.Errorf

same below

Reply:
we should stop here, there is no point in checking anything else. I prefer t.Fatal rather than e.Error followed by an empty return statement.

Reply:
t.Fatalf means that if we extend the test and will do something after this techCheckResultFunc this will no longer fire in case of error here

Reply:
But I don't feel super strongly about it - I can live with t.Fatalf here.